docs/reference/api/resources/list_source.rst

```diff
@@ -0,0 +1,80 @@
+ListSource
+==========
+
+A data source describing an ordered list of data.
+
+.. rst-class:: widget-support
+.. csv-filter:: Availability (:ref:`Key <api-status-key>`)
+   :header-rows: 1
+   :file: ../../data/widgets_by_platform.csv
+   :included_cols: 4,5,6,7,8,9
+   :exclude: {0: '(?!(ListSource|Resource))'}
+
+Usage
+-----
+
+Data sources are abstractions that allow you to define the data being managed by
+your application independent of the GUI representation of that data.
+
+ListSource is an implementation of an ordered list of data. When a ListSource is
+created, it is given a list of ``accessors`` - these are the attributes that
+all items managed by the ListSource will have.
+
+.. code-block:: python
+
+    from toga.sources import ListSource
+
+    source = ListSource(
+        accessors=["name", "weight"],
+        data=[
+            {"name": "Platypus", "weight": 2.4},
+            {"name": "Numbat", "weight": 0.597},
+            {"name": "Thylacine", "weight": 30.0},
+        ]
+    )
+
+    # Get the first item in the source
+    item = source[0]
```

Comment:
Mostly agreed on all these points; I guess my question is whether we tackle that immediately, or as a follow up "make ListSource a MutableSequence" PR. My inclination is to defer for now, as we can fix Sources independently to the audit of Selection.

Reply:
On second thought - it's not as big a change as I originally thought. I've made this change.

Reply:
Closing this out: AFAICT, `index()` and `remove()` are correct as is - they have to use row instances, because we can't use attribute equality comparisons. If I have a ListSource tracking 2 rows that have the same data value, they have to be treated as 2 distinct rows. On that basis, `remove(row)` and `index(row)` both need to use the row instance for lookup. `find()` exists to help find a row based on data.